### DISCUSSION TITLE
Test script for multiple inf files

### DISCUSSION THREAD

**Discussion by ANAT01** [lwcorp/lwmenu#10] [Show and tell]
For test multiple INF files I write batch script which list "inf" folder and run lwmenu with autorun.inf in selected folder.

```batch
@echo off

setlocal EnableDelayedExpansion

set "inf_dir=%~dp0INF"
set "count=0"

echo Current script path is: %~dp0
echo Current computer name is: %COMPUTERNAME%

echo The following folders are available in %inf_dir%:
for /d %%f in ("%inf_dir%\*") do (
	set /A "count+=1"
    echo #!count! - %%~nxf
)

set /p choice=Enter the number of the folder to run test:

set "count=0"
for /d %%f in ("%inf_dir%\*") do (
    set /A "count+=1"
    if !count!==!choice! (
        %~dp0lwmenu.exe -ini=%%~ff
        exit /b
    )
    REM echo CHOISE IS !choice! AND COUNT is !count! AND FOLDER IS %%~ff
)

pause
```

My folder structure is:

![изображение](https://github.com/lwcorp/lwmenu/assets/7475811/c70893c8-e612-473e-aa9c-3e5fe8dbac32)

Script output of test.bat is:

![изображение](https://github.com/lwcorp/lwmenu/assets/7475811/ced9afee-124d-444d-880d-728e46f96d4a)

**Comment by lwcorp** [comment]
Cool use! Of course, do know that you can have a central menu (in an upper folder) that can open secondary ones in sub-folders.
It's shown in the default settings' file:
![image](https://github.com/lwcorp/lwmenu/assets/1773306/a46d2cbd-2df3-412b-af16-48d7ffd833b5)

**Comment by ANAT01** [comment]
You right. I know.
But I use it dynamicly for different PCs (POS-terminals) for checking/modifing buttons.
Each folder is it hostname of PC.

In my use case:
1. Copy/Sync full folder with lwmenu on PC
2. Create sub-folder in inf/ named like current hostname
3. Test it in test.bat
4. Run install.bat script to add command in regedit with using variables like CurrentScriptPath a ComputerName

**Reply by lwcorp** [reply]
You certainly provide the most interesting use cases. 🥇 
I've changed this discussion category's accordingly and also promoted it in the brand new section [special use cases](https://github.com/lwcorp/lwmenu#special-use-cases).

Will it be alright to close this discussion now?
